pkg/azurefile/azurefile.go

```diff
@@ -872,6 +883,85 @@ func (d *Driver) ResizeFileShare(ctx context.Context, subsID, resourceGroup, acc
 	})
 }
 
+// CopyFileShare copies a fileshare in the same storage account
+func (d *Driver) CopyFileShare(ctx context.Context, req *csi.CreateVolumeRequest, accountKey string, shareOptions *fileclient.ShareOptions, storageEndpointSuffix string) error {
+	if shareOptions.Protocol == storage.EnabledProtocolsNFS {
+		return fmt.Errorf("protocol nfs is not supported for volume cloning")
+	}
+	var volumeID string
+	if req.GetVolumeContentSource() != nil && req.GetVolumeContentSource().GetVolume() != nil {
+		volumeID = req.GetVolumeContentSource().GetVolume().GetVolumeId()
+	}
+	resourceGroupName, accountName, srcFileShareName, _, _, subsID, err := GetFileShareInfo(volumeID)
+	dstFileShareName := shareOptions.Name
+	if err != nil {
+		return status.Error(codes.NotFound, err.Error())
+	}
+	klog.V(2).Infof("begin to copy fileshare %s to %s", srcFileShareName, dstFileShareName)
+	if srcFileShareName == "" {
+		return fmt.Errorf("srcFileShareName is empty")
+	}
+	if dstFileShareName == "" {
+		return fmt.Errorf("dstFileShareName is empty")
+	}
+
+	klog.V(2).Infof("generate sas token for account(%s)", accountName)
+	accountSasToken, genErr := generateSASToken(accountName, accountKey, storageEndpointSuffix, d.sasTokenExpirationMinutes)
+	if genErr != nil {
+		return genErr
+	}
+
+	if _, rerr := d.cloud.GetFileShare(ctx, subsID, resourceGroupName, accountName, srcFileShareName); rerr != nil {
+		return fmt.Errorf("srcFileShareName(%s) get error: %s", srcFileShareName, rerr.Error())
+	}
+
+	timeAfter := time.After(waitForCopyTimeout)
+	timeTick := time.Tick(waitForCopyInterval)
+	srcPath := fmt.Sprintf("https://%s.file.%s/%s%s", accountName, storageEndpointSuffix, srcFileShareName, accountSasToken)
+	dstPath := fmt.Sprintf("https://%s.file.%s/%s%s", accountName, storageEndpointSuffix, dstFileShareName, accountSasToken)
+
+	jobid, status, err := d.getAzcopyJob(dstFileShareName)
```

Comment:
what about just using one function `getAzcopyJob(dstFileShareName)` to get jobid, job status directly, then you don't need to use `parseAzcopyJobSummary` func, and `getAzcopyJob(dstFileShareName)`  could return (status, error), if job is not found, return job not started status, you could define the enum of all kinds of job status, e.g. not started, running, complete.

Reply:
I enum `AzcopyJobState` and retrurn (AzcopyJobState, percent, error) in `getAzcopyJob(dstFileShareName)`